### PR TITLE
#156 implementation of get_user_groups functions

### PR DESCRIPTION
**Pull request by matalie**
Implement get all user groups function
Implement get user group of user function
Write tests for new implemented functions

### CODE REVIEW

File: test/test_db/test_db.py [hunk 1]
```diff
@@ -247,6 +247,51 @@ def test_function_create_new_user_group(self):
...
+        create_test_user_group_admin = f'''CREATE ROLE user_group_mobydq_admin;'''
+        self.connection.execute(create_test_user_group_admin)
+
+        # Call tested function to get all user groups
+        call_test_get_all_user_groups = f'''SELECT base.get_all_user_groups();'''
+        cursor = self.connection.execute(call_test_get_all_user_groups)
+        user_group_list = list(cursor)
+
+        # Assert all created user groups are in the returned list
+        self.assertTrue('user_group_mobydq' in [item[0] for item in user_group_list])
```
Comment (ThomasGassmann) [inline test/test_db/test_db.py]: Don't do the same list comprehension twice. Just put it into a variable.

File: test/test_db/test_db.py [hunk 2]
```diff
@@ -247,6 +247,51 @@ def test_function_create_new_user_group(self):
         self.assertEqual(row_data_source_policy[0], 'SELECT')
         self.assertEqual(row_data_source_policy_admin[0], 'ALL')
 
+    def test_function_get_all_user_groups(self):
+        # Insert test get all user group roles
+
+        # Prepare database, delete all user group roles in the pg_authid table
+        delete_user_groups_roles = f'''DELETE FROM pg_authid WHERE rolname LIKE 'user_group_%';'''
```
Comment (ThomasGassmann) [inline test/test_db/test_db.py]: Why f-string?

File: test/test_db/test_db.py [hunk 3]
```diff
@@ -247,6 +247,51 @@ def test_function_create_new_user_group(self):
...
+
+        # Create new user groups
+        create_test_user_group_standard = f'''CREATE ROLE user_group_mobydq;'''
```
Comment (ThomasGassmann) [inline test/test_db/test_db.py]: Why f-string?

File: test/test_db/test_db.py [hunk 4]
```diff
@@ -247,6 +247,51 @@ def test_function_create_new_user_group(self):
...
+        self.connection.execute(create_test_user_group_standard)
+        create_test_user_group_admin = f'''CREATE ROLE user_group_mobydq_admin;'''
```
Comment (ThomasGassmann) [inline test/test_db/test_db.py]: Why f-string?

File: test/test_db/test_db.py [hunk 5]
```diff
@@ -247,6 +247,51 @@ def test_function_create_new_user_group(self):
...
+
+        # Assert all created user groups are in the returned list
+        self.assertTrue('user_group_mobydq' in [item[0] for item in user_group_list])
+        self.assertTrue('user_group_mobydq_admin' in [item[0] for item in user_group_list])
+
+    def test_function_get_all_user_groups_by_user(self):
+        # Insert test get all user group roles of user
+
+        # Prepare database, delete test user and test user group role
+        delete_user_role = f'''DELETE FROM pg_authid WHERE rolname LIKE 'user_mobydq';'''
```
Comment (ThomasGassmann) [inline test/test_db/test_db.py]: Why f-string?

File: test/test_db/test_db.py [hunk 6]
```diff
@@ -247,6 +247,51 @@ def test_function_create_new_user_group(self):
...
+        self.connection.execute(delete_user_role)
+        delete_user_group_role = f'''DELETE FROM pg_authid WHERE rolname LIKE 'user_group_mobydq';'''
```
Comment (ThomasGassmann) [inline test/test_db/test_db.py]: Why f-string?

File: test/test_db/test_db.py [hunk 7]
```diff
@@ -247,6 +247,51 @@ def test_function_create_new_user_group(self):
...
+        self.connection.execute(delete_user_group_role)
+
+        # Create new user and a new user group role, grant user group to user
+        create_new_user = f'''CREATE ROLE user_mobydq;'''
```
Comment (ThomasGassmann) [inline test/test_db/test_db.py]: Why f-string?

File: test/test_db/test_db.py [hunk 8]
```diff
@@ -247,6 +247,51 @@ def test_function_create_new_user_group(self):
...
+        self.connection.execute(create_new_user)
+        create_new_user_group = f'''CREATE ROLE user_group_mobydq;'''
```
Comment (ThomasGassmann) [inline test/test_db/test_db.py]: Why f-string?

File: test/test_db/test_db.py [hunk 9]
```diff
@@ -247,6 +247,51 @@ def test_function_create_new_user_group(self):
...
+        self.connection.execute(create_new_user_group)
+        grant_user_group_roles_to_user = f'''GRANT user_group_mobydq TO user_mobydq;'''
```
Comment (ThomasGassmann) [inline test/test_db/test_db.py]: Why f-string?

File: test/test_db/test_db.py [hunk 10]
```diff
@@ -247,6 +247,51 @@ def test_function_create_new_user_group(self):
...
+        self.connection.execute(grant_user_group_roles_to_user)
+
+        call_test_get_all_user_groups_by_user = f'''SELECT base.get_all_user_groups_by_user('user_mobydq');'''
```
Comment (ThomasGassmann) [inline test/test_db/test_db.py]: Why f-string?

File: test/test_db/test_db.py (hunk 2 above)
Comment (ThomasGassmann) [reply test/test_db/test_db.py]: Why three quotes?

File: test/test_db/test_db.py (hunk 3 above)
Comment (ThomasGassmann) [reply test/test_db/test_db.py]: Why three quotes?

File: test/test_db/test_db.py (hunk 4 above)
Comment (ThomasGassmann) [reply test/test_db/test_db.py]: Why three quotes?

File: test/test_db/test_db.py [hunk 11]
```diff
@@ -247,6 +247,51 @@ def test_function_create_new_user_group(self):
...
+        delete_user_groups_roles = f'''DELETE FROM pg_authid WHERE rolname LIKE 'user_group_%';'''
+
+        # Create new user groups
+        create_test_user_group_standard = f'''CREATE ROLE user_group_mobydq;'''
+        self.connection.execute(create_test_user_group_standard)
+        create_test_user_group_admin = f'''CREATE ROLE user_group_mobydq_admin;'''
+        self.connection.execute(create_test_user_group_admin)
+
+        # Call tested function to get all user groups
+        call_test_get_all_user_groups = f'''SELECT base.get_all_user_groups();'''
```
Comment (ThomasGassmann) [inline test/test_db/test_db.py]: Why f-strings? Why three quotes?

File: test/test_db/test_db.py (hunk 5 above)
Comment (ThomasGassmann) [reply test/test_db/test_db.py]: Why three quotes?

File: test/test_db/test_db.py (hunk 6 above)
Comment (ThomasGassmann) [reply test/test_db/test_db.py]: Why three quotes?

File: test/test_db/test_db.py (hunk 7 above)
Comment (ThomasGassmann) [reply test/test_db/test_db.py]: Why three quotes?

File: test/test_db/test_db.py (hunk 8 above)
Comment (ThomasGassmann) [reply test/test_db/test_db.py]: Why three quotes?

File: test/test_db/test_db.py (hunk 9 above)
Comment (ThomasGassmann) [reply test/test_db/test_db.py]: Why three quotes?

File: test/test_db/test_db.py (hunk 10 above)
Comment (ThomasGassmann) [reply test/test_db/test_db.py]: Why three quotes?

File: test/test_db/test_db.py (hunk 8 above)
Comment (matalie) [reply test/test_db/test_db.py]: Changed both

File: test/test_db/test_db.py (hunk 1 above)
Comment (matalie) [reply test/test_db/test_db.py]: I changed to variable

File: test/test_db/test_db.py (hunk 2 above)
Comment (matalie) [reply test/test_db/test_db.py]: Changed both

File: test/test_db/test_db.py (hunk 3 above)
Comment (matalie) [reply test/test_db/test_db.py]: Changed both

File: test/test_db/test_db.py (hunk 4 above)
Comment (matalie) [reply test/test_db/test_db.py]: Changed

File: test/test_db/test_db.py (hunk 6 above)
Comment (matalie) [reply test/test_db/test_db.py]: Changed

File: test/test_db/test_db.py (hunk 7 above)
Comment (matalie) [reply test/test_db/test_db.py]: Changed both

File: test/test_db/test_db.py (hunk 9 above)
Comment (matalie) [reply test/test_db/test_db.py]: Changed both

File: test/test_db/test_db.py (hunk 10 above)
Comment (matalie) [reply test/test_db/test_db.py]: Changed both

File: test/test_db/test_db.py (hunk 11 above)
Comment (matalie) [reply test/test_db/test_db.py]: Changed both

**Pull request merged**